### DISCUSSION TITLE
Invalidation step report

### DISCUSSION THREAD

**Discussion by ChristopheBordieu** [oracle/opengrok#4029] [Q&A] [answered]
Hi team

In 1.7.35 logs, I can find twice a log regarding invalidation of repositories:
```
...
2022-08-17 08:01:26.189+0000 INFO t1 Statistics.logIt: Done invalidating repositories (1249 valid, 1249 working) (took 5.351 seconds)
...
2022-08-17 08:02:20.118+0000 INFO t1 Statistics.logIt: Done invalidating repositories (1249 valid, 1249 working) (took 2.632 seconds)
...
```

Is it expected to have twice these "same" ines before and after history cache actions?
What invalidation step is actually? What "valid" vs "working" mean?

**Comment by vladak** [comment, marked as answer]
Is this for the webapp or indexer ?

**Reply by ChristopheBordieu** [reply]
Hi @vladak 

It is for indexer.
I am reporting some data each time I rhave run indexer grepping the opengrok logs.
And actually I don't clearly understand what these messages mean so my question.

**Reply by vladak** [reply]
Firstly, the invalidation of repositories during indexer run is done only once, at least in the current version. The invalidation happens whenever a new configuration is set. As for the classification, "valid" means that the related directory was identified as a repository (e.g. Git), "working" has a a bit of a broad/vague meaning (that can be also overridden for given repository type) however basically it means that it is possible to interact with the repository (e.g. query the history).

**Reply by ChristopheBordieu** [reply]
Thanks for the answers!
